includes/classes/CouponValidation.php

```diff
@@ -32,6 +32,12 @@ public static function is_product_valid(int $product_id, int $coupon_id): bool
             return false;
         }
 
+        $product_can_use_coupon = true; 
+        $this->notify('NOTIFY_COUPON_ADDITIONAL_CHECKS', $product->fields, $coupon_id, $product_can_use_coupon); 
```

Comment:
`$this->notify` is not available in this class.

Will have to use $zco_notifier instead:
`global $zco_notifier;` at the top of the function
and use `$zco_notifier->notify` instead of `$this->notify`

Reply:
geez I thought i had fixed that.  Sorry.

Reply:
Pretty sure that's why the test is failing: syntax error causing blank page, so test is looking for a product price in a blank string.

Reply:
Yep.  I implemented this in a 2.0 upgrade and thought I had copied my work to the github branch, but I think I fell behind. 

Fixed now.